### PR TITLE
Polish custom layout implementation

### PR DESCRIPTION
**Pull request by samtstern**
Some final items for #1477 

TODO:

- [x] Make a nice custom layout for the sample app
- [x] Make the custom layout chosen with a radio button in the sample app
- [x] Obvious failure for provider configuration mismatch
- [x] README
- [x] JavaDoc
- [x] ToS and PP in custom layout

Here's a preview:
device-2018-10-31-113747

### CODE REVIEW

**Comment by samtstern** [comment]
cc @iamyaoxi

**Comment by iamyaoxi** [comment]
The layout is very sweet!!

File: app/src/main/res/drawable/custom_bg_gradient.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android">
+    <gradient
+        android:startColor="#7986cb"
+        android:endColor="#7e57c2"
+        android:angle="270"/>
+</shape>
```
Comment (lsirac) [inline app/src/main/res/drawable/custom_bg_gradient.xml]: Nit: no new line

File: app/src/main/res/layout/auth_method_picker_custom_layout.xml [hunk 2]
```diff
@@ -1,57 +1,146 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/header_idp_choice"
+        app:layout_constraintTop_toTopOf="@+id/header_idp_choice">
+
+        <View style="@style/CustomWingInner" />
+
+    </FrameLayout>
+
+    <!--
+      NOTE: This sample app uses this class from the FirebaseUI library to show the Google Sign
```
Comment (lsirac) [inline app/src/main/res/layout/auth_method_picker_custom_layout.xml]: I don't understand - what's wrong with using the Google sign in button?

**Pull request merged**